spec/tapioca/compilers/dsl/active_record_columns_spec.rb

```diff
@@ -0,0 +1,336 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+require "tapioca/compilers/dsl/active_record_columns"
+
+RSpec.describe(Tapioca::Compilers::Dsl::ActiveRecordColumns) do
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecord subclasses") do
+      expect(subject.processable_constants).to(be_empty)
+    end
+
+    it("gathers only ActiveRecord subclasses") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+        end
+
+        class Current
+        end
+      RUBY
+
+      expect(constants_from(content)).to(eq(["Post"]))
+    end
+
+    it("rejects abstract ActiveRecord subclasses") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+        end
+
+        class Current < ActiveRecord::Base
+          self.abstract_class = true
+        end
+      RUBY
+
+      expect(constants_from(content)).to(eq(["Post"]))
+    end
+  end
+
+  describe("#decorate") do
+    before(:all) do
+      ActiveRecord::Base.establish_connection(
+        adapter: 'sqlite3',
+        database: ':memory:'
+      )
+    end
+
+    def rbi_for(contents)
+      with_contents(contents, requires: contents.keys) do
+        parlour = Parlour::RbiGenerator.new(sort_namespaces: true)
+        subject.decorate(parlour.root, Post)
+        parlour.rbi
+      end
+    end
+
+    def indented(str, indent)
+      str.lines.map! do |line|
+        next line if line.chomp.empty?
+        " " * indent + line
+      end.join
+    end
+
+    it("generates RBI file for class without custom attributes with StrongTypeGeneration") do
```

Comment:
I didn't test this case "without" StrongTypeGeneration. 

Tested "classes with custom attributes with/without StrongTypeGeneration" down below. I thought it would add another long test without providing much value but I can add it.